File: backend-java/src/main/java/com/myforum/model/entity/Post.java
package com.myforum.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("post")
public class Post {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private Integer categoryId;
    private String title;
    private String content;
    private String contentRaw;
    private Integer status;     // 1-正常 0-删除 2-置顶
    private Long viewCount;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableField(exist = false)
    private User user;
    @TableField(exist = false)
    private Category category;
    @TableField(exist = false)
    private Long likeCount;
    @TableField(exist = false)
    private Boolean liked;
    @TableField(exist = false)
    private Boolean favorited;
}

File: frontend/src/api/post.ts
import request from './request'

export interface PostItem {
  id: number
  title: string
  content: string
  content_raw?: string
  status: number
  view_count: number
  like_count: number
  favorite_count: number
  category_id: number
  user_id: number
  created_at: string
  updated_at: string
  user?: { id: number; username: string; avatar: string }
  category?: { id: number; name: string; slug: string }
  liked?: boolean
  favorited?: boolean
}

export interface PostListResult {
  list: PostItem[]
  total: number
  page: number
  page_size: number
}

export interface PostQuery {
  page?: number
  page_size?: number
  category_id?: number
  sort?: string
  keyword?: string
  user_id?: number
}

export const postAPI = {
  list(params?: PostQuery): Promise<PostListResult> {
    return request.get('/posts', { params })
  },

  getDetail(id: number): Promise<PostItem> {
    return request.get(`/post/${id}`)
  },

  create(data: { category_id: number; title: string; content: string }) {
    return request.post('/post', data)
  },

  update(id: number, data: { title?: string; content?: string }) {
    return request.put(`/post/${id}`, data)
  },

  delete(id: number) {
    return request.delete(`/post/${id}`)
  },

  search(params: { keyword: string; page?: number; page_size?: number }): Promise<PostListResult> {
    return request.get('/posts/search', { params })
  },
}
